nistats/first_level_model.py

```diff
@@ -538,26 +538,31 @@ def compute_contrast(self, contrast_def, stat_type=None,
         if len(con_vals) != n_runs:
             warn('One contrast given, assuming it for all %d runs' % n_runs)
             con_vals = con_vals * n_runs
-        if isinstance(output_type, _basestring):
-            if output_type not in ['z_score', 'stat', 'p_value', 'effect_size',
-                                   'effect_variance']:
-                raise ValueError('output_type must be one of "z_score", "stat",'
-                                 ' "p_value","effect_size" or "effect_variance"')
-        else:
-            raise ValueError('output_type must be one of "z_score", "stat",'
-                             ' "p_value","effect_size" or "effect_variance"')
+
+        valid_types = ['z_score', 'stat', 'p_value', 'effect_size',
+                       'effect_variance']
```

Comment:
Why not add `'all'` to the list here instead of doing a `+ ['all']` later?

Reply:
Mostly because when the actual value is `'all'`, we want access to just this list of 5 things.

```Python
output_types = valid_types if output_type == 'all' else [output_type]
```

We could set `valid_types = [..., 'all']` and then say `output_types = valid_types[:-1] if ...`, but it's not clearly improved. Whichever you prefer is fine with me.